.github/workflows/build-windows-x86_64.yml

```diff
@@ -86,14 +90,53 @@ jobs:
       # Use a retry action because inexplicably sometimes stack install stalls
       # indefinitely after the `compiling Main` step, but then succeeds immediately
       # with a second stack install. 🤷🏻‍♂️
-      - uses: nick-invision/retry@v2
+      - uses: nick-fields/retry@v2
         with:
           timeout_minutes: 5
-          max_attempts: 3
-          command: stack install
+          max_attempts: 4
+          # https://gitlab.haskell.org/ghc/ghc/-/issues/20878
+          # https://gitlab.haskell.org/ghc/ghc/-/issues/20010#note_359766
+          command: stack install --ghc-options '-optl"-Wl,-Bstatic,-lstdc++,-lgcc_s,-lwinpthread,-Bdynamic"'
 
-      - name: Archive elm-dev.exe artifact
-        uses: actions/upload-artifact@v2
+      - name: Copy binary files, dlls & check
+        run: |
+          mkdir elm-dev
+          cp ~/AppData/Roaming/local/bin/elm-dev.exe elm-dev/elm-dev.exe
+          cp distribution/dlls/* elm-dev/
+          ./elm-dev/elm-dev.exe --version-full
+
+      - name: Create distribution zip
+        uses: vimtor/action-zip@v1
         with:
-          name: elm-dev.exe
-          path: ~\AppData\Roaming\local\bin\elm-dev.exe
+          files: elm-dev/
+          recursive: false
+          dest: elm-dev.zip
+
+      - name: Archive elm-dev.zip artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: elm-dev-next.zip
+          path: elm-dev.zip
+
+  upload:
+    needs: install
+    name: Upload elm-dev-next.zip to SFTP
+    runs-on: lamdera-community-linux-x86_64
+    steps:
+      - name: Download artifacts
+        uses: actions/download-artifact@v3
+        with:
+          name: elm-dev-next.zip
+
+      - name: Install SSH key
+        uses: shimataro/ssh-key-action@v2
+        with:
+          key: ${{ secrets.GH_USER_SCP_KEY }}
+          name: id_ed25519
+          known_hosts: ${{ secrets.KNOWN_HOSTS }}
+          if_key_exists: replace
+
+      - run: |
+          pwd
+          ls -alh
+          echo "put elm-dev.zip next/elm-dev-next-windows-x86_64.zip" | sftp -i ~/.ssh/id_ed25519 -P 22 github@apps.lamdera.com
```

Comment:
I currently do this for all Lamdera builds because I've had issues with the Artifacts caching getting exhausted in the Github free tier and being a general annoyance to manage manually with builds failing until you clear it out. Given the `next` builds are basically "nightly" builds, in the Lamdera context it makes to have a single output that keeps being replaced until we cut a new version. Could keep the same setup for elm-dev with a different target, what do you think?

Reply:
Yeah, I like that a lot, makes it really easy to share when things are being developed.